src/ship/CameraController.cpp

```diff
@@ -219,29 +219,21 @@ void ExternalCameraController::Update()
 	m_extOrient = matrix3x3d::RotateY(-m_rotY) *
 		matrix3x3d::RotateX(-m_rotX);
 
-	SetPosition(p);
-	SetOrient(m_extOrient);
+	const matrix3x3d &shipOrient = GetShip()->GetInterpOrient();
 
-	m_camera->SetCameraFrame(GetShip()->GetFrame());
-	if (GetType() == FLYBY) {
-		m_camera->SetCameraOrient(GetOrient());
-		m_camera->SetCameraPosition(GetPosition());
-	} else {
-		// This lerping factor feels nice and scales non-linearly with larger (and slower-turning) ships
-		float lerp_factor = (2.5 + GetShip()->GetClipRadius() * 0.05) * Pi::GetFrameTime();
-		// Smooth the ship orientation by lerping toward the current orientation
-		m_smoothed_ship_orient = Quaternionf::lerp(
-			m_smoothed_ship_orient,
-			Quaternionf::FromMatrix3x3(GetShip()->GetInterpOrient()),
-			lerp_factor)
-									 .Normalized();
-		const matrix3x3d &smoothed_m = m_smoothed_ship_orient.ToMatrix3x3<double>();
+	// This lerping factor feels nice and scales non-linearly with larger (and slower-turning) ships
+	float lerp_factor = (2.5 + GetShip()->GetClipRadius() * 0.05) * Pi::GetFrameTime();
+	// Smooth the ship orientation by lerping toward the current (frame-invariant) orientation
+	Quaternionf frame_quat = Quaternionf::FromMatrix3x3(Frame::GetFrame(GetShip()->GetFrame())->GetOrient());
```

Comment:
This line should be using `GetOrientRelTo(Pi::game->GetSpace()->GetRootFrame())` - it's very possible there are several frames in between the ship's frame and the root frame, which will make this calculation incorrect.

In the case that the root frame is the parent of the ship's frame, this is the same as `GetOrient()` so the frame-invariant behavior is preserved.

Reply:
Changed!